src/main/java/lesson2/problem4/DeleteMiddleNode.java

```diff
@@ -0,0 +1,41 @@
+package lesson2.problem4;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+public class DeleteMiddleNode {
+    // O(n) time | O(1) space
+    // Not sure why slides say O(1) time... we must iterate through nodes
+    public static void deleteMiddleNode (Link head) {
```

Comment:
Argument name should be "node", using "head" is confusing as it might refer to the beginning of the list.